### PR TITLE
Fix some old JSON characters not loading

### PR DESCRIPTION
**Pull request by mcmonkey4eva**
Fixes bug introduced @ https://github.com/oobabooga/text-generation-webui/pull/337

tldr: if a `json` file has a tab in it, it breaks. With this PR it no longer breaks as it uses the JSON parser for `json` files, and YAML parser for `yaml`/`yml` files. This should also preemptively fix any other format edgecases that might exist between the two.

I also improved the Example yaml character file format as a secondary commit. Hopefully you'll agree with me that it looks a lot cleaner this way, and a lot friendlier to use as a reference for new users.
Specifically it uses multiline-block syntax (that lil `|-`) to automatically remove the need for quotes and to make a single newline count as a newline (to avoid the spam of double newlines). Also use double quotes `"` in the `context` key to avoid the need for that awkward `''` double-singlequote escape syntax.

### CODE REVIEW

**Comment by oobabooga** [comment]
You are absolutely right that it's better to load json with the json loader and yaml with the yaml loader. I did not realize that there could be conflicts because all my characters just worked.

I reverted the change to the yaml example because the `vim` syntax highlight didn't work properly with this new formatting.

**Comment by mcmonkey4eva** [comment]
... *vim*?!

**Comment by oobabooga** [comment]
It's all I use :P

**Comment by TheTerrasque** [comment]
Bless your heart 😄 

But on a more serious note, the example character should have the more newbie friendly layout, I'm sure people who has found out how to use vim can handle some lack of syntax highlighting.

**Comment by TheTerrasque** [comment]
yaml has many ways to handle blocks of text, actually: https://yaml-multiline.info/

**Comment by mcmonkey4eva** [comment]
Yay!
 #770
